packages/tunnel-client/src/lib/tunnel-multiplexing-cluster.ts

```diff
@@ -141,21 +175,23 @@ export class TunnelMultiplexingCluster extends (EventEmitter as new () => TypedE
       });
     };
 
-    remote.on("data", (data: any) => {
-      // parse the first (request) line of the request to determine the method and path
-      // Example: GET /path HTTP/1.1
-      const match = data.toString().match(/^(\w+) (\S+)/);
-      if (match) {
-        this.emit("request", {
-          method: match[1],
-          path: match[2],
-        });
-      }
-    });
+    this.remoteMuxClient.on("handshake", (stream) => {
+      stream.on("data", (data: any) => {
+        // parse the first (request) line of the request to determine the method and path
+        // Example: GET /path HTTP/1.1
+        const match = data.toString().match(/^(\w+) (\S+)/);
```

Comment:
It's probably fine because the format here is likely to be very fixed, but just in case should we make this a bit more robust and handle whitespace at the start of the line and extra spaces between the method and path?

Reply:
Looks like the spec doesn't allow for trailing whitespaces so I think this should be fine -- this is also used for debug printing of requests when `--printRequests` is provided